src/app/table/table.component.ts

```diff
@@ -1,32 +1,53 @@
 import { Component, OnInit } from '@angular/core';
-import { ResponseModel } from '../models/response-model';
-import { PostService } from '../services/post.service';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs';
+import { ResponseModel } from 'src/app/models/response-model';
+import { postRequest } from 'src/app/store/action-store';
+import { AppState } from 'src/app/store/app-state-model';
+
+
 
 @Component({
   selector: 'app-table',
   templateUrl: './table.component.html',
   styleUrls: ['./table.component.css']
 })
 export class TableComponent implements OnInit {
-  bothPureAndPalindrome: ResponseModel[] = [];
-  onlyPalindrome: ResponseModel[] = [];
-  notPalindrome: ResponseModel[] = [];
 
-  constructor(private postService: PostService) { }
+  bothPureAndPalindrome$: ResponseModel[] = [];
+  onlyPalindrome$: ResponseModel[] = [];
+  notPalindrome$: ResponseModel[] = [];
+
+  private bothPureAndPalindromeSub: Subscription = new Subscription();
+  private onlyPalindromeSub: Subscription = new Subscription();
+  private notPalindromeSub: Subscription = new Subscription();
 
-  ngOnInit(): void {
-  }
 
-  postData() {
+  constructor(private store: Store<AppState>) {
+  }
 
-    this.postService.postData().subscribe((res: any) => {
-      this.bothPureAndPalindrome = res.filter((resBody: ResponseModel) => resBody.is_pure_palindrome);
-      this.onlyPalindrome = res.filter((resBody: ResponseModel) => !resBody.is_pure_palindrome && resBody.is_palindrome);
-      this.notPalindrome = res.filter((resBody: ResponseModel) => !resBody.is_palindrome);
-      console.log("Pure : ", this.bothPureAndPalindrome);
-      console.log("OnlyPalindrome : ", this.onlyPalindrome);
-      console.log("Not : ", this.notPalindrome);
+  ngOnInit(): void {
+    const payload = {
+      content: ["racecar",
+        "A man, a plan, a canal: Panama.",
+        "A test"]
+    };
+    this.store.dispatch(postRequest({ payload }));
 
+    this.bothPureAndPalindromeSub = this.store.select(state => state.post.bothPureAndPalindrome).subscribe(data => {
```

Comment:
her birini tek tek subscription'a eşitlemektense bir subscription array'i tutup onInit'te subscriptionları tetiklemek daha mantıklı.  this.store.select'i return eden bir fn olmalı.

subscriptions: Subscription[]; 

ngOnInit(){
this.subscribeToData()
}

subscribeToData(){
this.subscriptions = [
this.subscribeToPurePalindrome() 
]
}

şeklinde